src/tox/session/commands/run/parallel.py

```diff
@@ -39,6 +39,10 @@ def run_in_thread(tox_env, os_env, processes):
                 if hasattr(tox_env, "package"):
                     args_sub.insert(position, str(tox_env.package))
                     args_sub.insert(position, "--installpkg")
+                if config.option.resultjson:
+                    result_json_index = args_sub.index('--result-json')
+                    result_json_path = tox_env.path.join('.tox-result.json').strpath
+                    args_sub[result_json_index + 1] = result_json_path
```

Comment:
use explicit string format instead please, this also avoids the need for the strpath above; define path upfront

Reply:
I would propose instead using https://github.com/tox-dev/tox/blob/master/src/tox/util/lock.py#L27 to get a unique file, otherwise this solution, for example, would break if two parallel operations are running in parallel

Reply:
Actually, the result json is user input, so we don't need the locked thing, the index is ok. Just make sure to save it on envconfig.

Reply:
oh nm 👍